redash/handlers/data_sources.py

```diff
@@ -166,7 +170,48 @@ def get(self, data_source_id):
         try:
             if refresh:
                 refresh_schema.apply_async(args=(data_source.id,), queue=settings.SCHEMAS_REFRESH_QUEUE)
-            response['schema'] = data_source.get_schema()
+
+            schema = []
+            columns_by_table_id = {}
+
+            tables = TableMetadata.query.filter(
+                TableMetadata.data_source_id == data_source.id,
+                TableMetadata.exists.is_(True),
+            ).options(joinedload(TableMetadata.sample_queries)).all()
+            table_ids = [table.id for table in tables]
+
+            columns = ColumnMetadata.query.filter(
+                ColumnMetadata.exists.is_(True),
+                ColumnMetadata.table_id.in_(table_ids),
+            ).all()
+
+            for column in columns:
+                columns_by_table_id.setdefault(column.table_id, []).append({
+                    'key': column.id,
+                    'name': column.name,
+                    'type': column.type,
+                    'exists': column.exists,
+                    'example': column.example,
+                    'description': column.description,
+                })
+
+            for table in tables:
+                sample_queries_dict = dict([(v['id'], v) for v in QuerySerializer(table.sample_queries).serialize()])
+                table_info = {
+                    'id': table.id,
+                    'name': table.name,
+                    'exists': table.exists,
+                    'visible': table.visible,
+                    'hasColumnMetadata': table.column_metadata,
+                    'description': table.description,
+                    'sample_queries': sample_queries_dict,
+                    'columns': []}
```

Comment:
It would be neat to have a `TableMetadataSerializer` and `ColumnMetadataSerializer` for this so we can separate serialization from app-logic.

Reply:
Done